reference/shaders-msl-no-opt/packing/matrix-3x2-scalar.comp

```diff
@@ -5,7 +5,7 @@
 
 using namespace metal;
 
-typedef packed_float3 packed_rm_float3x2[2];
+typedef float2x3 packed_rm_float3x2;
```

Comment:
Still wrong here with packing.

Reply:
Overall, would it be okay to simply remove any use of spvUnsafeArray for any buffer-related arrays?

Reply:
Sorry about that, I fixed that for the next commit.

Reply:
I mean arrays which are declared inside buffers: `struct UBO { float4 arr[16]; }; kernel void main(constant UBO &ubo) {}`.

The struct type has "Offset" decorations if it's used as a "buffer-like" type, so I think that might be usable.

From what I understand, spvUnsafeArray is only really used for variables outside buffer objects to improve copying arrays around as value-types, but copying array types from buffers doesn't really happen afaik. At any rate, it's not well tested.

If there is a reliable way to get `spvUnsafeArray<packed_float3, N>` emitted instead without too much hackery, that might be an option, but it doesn't seem easy.